vsintegration/src/FSharp.Editor/FSharpBraceMatchingService.fs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.Composition
+open System.Collections.Concurrent
+open System.Collections.Generic
+open System.Threading
+open System.Threading.Tasks
+open System.Linq
+
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.Classification
+open Microsoft.CodeAnalysis.Editor
+open Microsoft.CodeAnalysis.Editor.Implementation.BraceMatching
+open Microsoft.CodeAnalysis.Editor.Shared.Utilities
+open Microsoft.CodeAnalysis.Host.Mef
+open Microsoft.CodeAnalysis.Text
+
+open Microsoft.VisualStudio.FSharp.LanguageService
+open Microsoft.VisualStudio.Text
+open Microsoft.VisualStudio.Text.Tagging
+
+open Microsoft.FSharp.Compiler.Parser
+open Microsoft.FSharp.Compiler.SourceCodeServices
+
+[<ExportBraceMatcher(FSharpCommonConstants.FSharpLanguageName)>]
+type internal FSharpBraceMatchingService() =
+              
+    static member private SupportedBraceTypes = [
+        ('(', ')');
+        ('<', '>');
+        ('[', ']');
+        ('{', '}');
+    ]
+    
+    static member private IgnoredClassificationTypes = [
+        ClassificationTypeNames.Comment;
+        ClassificationTypeNames.StringLiteral;
+        ClassificationTypeNames.ExcludedCode;
+    ]
+
+    interface IBraceMatcher with
+        member this.FindBracesAsync(document: Document, position: int, cancellationToken: CancellationToken): Task<Nullable<BraceMatchingResult>> =
+            let computation = async {
```

Comment:
will this do an implicit ConfigureAwait(false)?   If not, can you change this to:  document.GetTextAsync(ct).ConfigureAwait(false)